pkg/apis/core/types.go

```diff
@@ -413,6 +413,14 @@ type PersistentVolumeClaimSpec struct {
 	// This is an alpha feature and may change in the future.
 	// +optional
 	VolumeMode *PersistentVolumeMode
+	// If specified, volume will be prepopulated with data from the specified data source.
+	// This depends on the provisioner for this volume being able to use the specified source.
+	// If the provisioner does not support it, it will create an empty volume.
```

Comment:
How do I know if the data init succeeded or not?  Does the provisioner also have to log and event (which disappears) or set an error somewhow?

Reply:
Added some comments to clarify this.

Reply:
if provisioner failed to create volume from the data source, the behavior will the same as no data source case (failed to create an empty volume), PV will not be created, and PVC will continue in pending state. The controller will keep retrying to create the volume